toc/rfc/rfc-draft-multiple-identity-zones.md

```diff
@@ -0,0 +1,51 @@
+# Meta
+[meta]: #meta
+- Name: User Account Management in Cloud Foundry with Multiple Identity Zones
+- Start Date: 2023-08-04
+- Authors: @FloThinksPi, @klaus-sap, @philippthun
+- Status: Draft
+- RFC Pull Request: https://github.com/cloudfoundry/community/pull/665
+
+## Summary
+
+This RFC proposes enhancing Cloud Foundry (CF) to support users from multiple identity zones within the CF User Account and Authentication (UAA) system. The objectives are to enable efficient user management, facilitate data isolation and allow customized security measures.
+
+## Problem
+
+CF operators are currently limited to the default identity zone within the UAA system for user account management. However, the need to store user accounts in different zones may arise from two distinct operational requirements:
+1. CF installations servicing multiple customers or tenants (e.g. public PAAS) must ensure absolute isolation of user accounts. Each customer or tenant must have their user accounts securely stored in separate identity zones to prevent unauthorized access to sensitive data. It is crucial that users from one customer cannot view, modify, or access the user accounts of other customers.
+1. Even in cases where CF is deployed for a single customer or organization, there may be a need to employ multiple identity zones. This could arise when customers wish to segregate user accounts based on roles, affiliations, or subsidiaries. By creating distinct zones, organizations can enforce specific security measures tailored to different user groups, such as implementing Multi-Factor Authentication (MFA) for certain privileged users.
+
```

Comment:
One additional problem that this RFC tries to solve: Customers want to have full control of their CF users, i.e. create users, assign CF roles, and finally also remove users. It should be possible for customers to implement a central user provisioning based on their needs.

This is not possible with a central UAA using the default identity zone only because we can't grant admin access for the default zone to customers. Admin access for a customer identity zone would solve this.

Reply:
I see. The use case is ultimately about having multiple organizations sharing the same CF foundation, where each of the orgs needs to self-manage their CF users.

Reply:
@peterhaochen47 The operational costs itself are imho a showstopper for org-specific UAA instances. Apart from that, the UAA integration with central landscape components (which need access to all UAAs) becomes overly complex and would most probably not scale.

Reply:
@torsten-sap how about this option:
> You might even be able to just add one non-system identity zone as [an external identity provider](https://docs.cloudfoundry.org/api/uaa/version/76.18.0/index.html#create) of the main system UAA zone (on the UAA instance). But I haven't verified that it works (but I don't see why not).

Reply:
yes but the resulted user token is then with zid:uaa means that users are within the system (uaa) zone, but we need and what separate users (and other related entities like idps or groups) into an own zones, so that a zone can be configured independently from another zone. That is what we support today in UAA, but not in cloud controller... 

The move of groups and IdPs into the customer zone is not a big challenge, but the oauth clients (e.g. existing dashboards https://docs.cloudfoundry.org/services/dashboard-sso.html ) and their configuration with secrets is something I think about improvements, so that not all dashboard clients need to be copied into each single zone, therefore we want extend the central check_token endpoint.  The details we should discuss in our call

Reply:
> It's similar to that when you log into uaa system zone via Azure AD and get a UAA token, the user is still located in Azure AD.

true, but then we have the same as today. What we want have is a separation of users so that different administrators can manage these different users. Then we would need a new administrator, the origin admin who only can manage users and groups for a certain origin.